src/main/java/com/mcmoddev/lib/API/IBlockApi.java

```diff
@@ -0,0 +1,23 @@
+package com.mcmoddev.lib.API;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import com.mcmoddev.lib.material.MMDMaterial;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.ItemBlock;
+
+public interface IBlockApi extends IMMDApi {
+	default String getApiName() {
+		return "MMDLib Block->Material Registration API";
+	}
+	
+	Pair<? extends Block, ? extends ItemBlock> createBlock( String typeName, MMDMaterial material);

```

Comment:
Not all blocks have ItemBlocks. How should this be handled?

Reply:
All blocks have an ItemBlock - a lot of them use the built-in base ItemBlock, though.